### PR TITLE
Kafka 1.4.3 upgrade

### PR DESCRIPTION
**Pull request by emord**
@dimagi/scale-team They changed the way they poll for changes in one of these upgrades that should hopefully the file handle errors we've been flooded with.

Not as hard as I thought it would be. Things that were broken were mostly just renamed or moved

https://kafka-python.readthedocs.io/en/master/changelog.html

### CODE REVIEW

**Comment by emord** [comment]
Tested this for a bit on staging. for both SQL and couch domains. I also tested shutting the pillow down, making a submission, and then starting hte pillow back up to ensure it was still using the same offsets

File: corehq/apps/change_feed/tests/test_pillow.py [hunk 1]
```diff
@@ -36,39 +36,16 @@ def setUp(self):
...
-    def test_process_change_with_unicode_domain(self):
-        document = {
-            'doc_type': 'CommCareCase',
-            'type': 'mother',
-            'domain': 'हिंदी',
-        }
-        self.pillow.process_change(Change(id='test-id', sequence_id='3', document=document))
-        message = next(self.consumer)
-        change_meta = change_meta_from_kafka_message(message.value)
-        self.assertEqual(document['domain'], change_meta.domain)
```
Comment (czue) [inline corehq/apps/change_feed/tests/test_pillow.py]: it wasn't totally obvious to me why these two tests were removed. are they redundant?

**Review by czue — COMMENTED** [review]
:fire:

File: corehq/apps/change_feed/tests/test_pillow.py (hunk 1 above)
Comment (emord) [reply corehq/apps/change_feed/tests/test_pillow.py]: I don't know. it was probably accidental when I was working on fixing them

**Comment by emord** [comment]
Changes full under the following buckets:

a) renames from library (TopicAndPartition -> TopicPartition, etc)
b) Changes in manually setting partitions (using assign instead of `set_topic_partitions`)
c) initializing test change feeds by reading message. required because feeds start at end of the queue, but the queue is intialized until you try to read a message/manually assign offsets.
c) using client_id instead of group_id. group_id is used to automatically assign partitions between consumers and would be really nice to use in the in the future, but kafka would need to be upgraded. Fun fact: our kafka docker server is not the same version we run in production. 0.8 vs 0.10

**Review by czue — APPROVED** [review]

**Pull request merged**